test/decl/ext/extensions.swift

```diff
@@ -153,3 +153,13 @@ extension DoesNotImposeClassReq_2 where Self : AnyObject {
     set { property = newValue } // Okay
   }
 }
+
+// Reject extension of nominal type via indirection through a typealias
```

Comment:
Type aliases are fine -- the problematic case here is a typealias whose underlying type is a type parameter. We can't resolve it at extension binding time without a conformance check.